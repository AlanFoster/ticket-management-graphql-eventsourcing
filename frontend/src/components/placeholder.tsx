import * as React from "react";
import {makeStyles} from "@material-ui/core";

const useSkeletonStyles = makeStyles(theme => ({
    root: {

    },
    '@keyframes skeleton-animation': {
        '0%': {
            backgroundPosition: '-200px 0'
        },
        '100%': {
            backgroundPosition: 'calc(200px + 100%) 0'
        }
    },
    element: {
        ...theme.typography.h3,
        height: '1em',
        width: '100%',
        display: 'inline-block',
        background: '#eee',
        backgroundImage: 'linear-gradient(90deg, #eee, #f5f5f5, #eee)',
        backgroundSize: '200px 100%',
        borderRadius: '4px',
        backgroundRepeat: 'no-repeat',
        animation: '$skeleton-animation 1.2s ease-in-out infinite'
    }
}));

export const Placeholder = function ({ count = 1 }) {
    const classes = useSkeletonStyles();
    const elements = [];
    for (var i = 0; i < count; i++) {
        elements.push(<span key={i} className={classes.element} />)
    }

    return (
        <div className={classes.root}>
            {elements}
        </div>
    );
};
