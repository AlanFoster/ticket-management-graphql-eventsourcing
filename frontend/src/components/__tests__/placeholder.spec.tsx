import * as React from "react";
import { render, cleanup } from "@testing-library/react";
import { Placeholder } from "../placeholder";

describe("Placeholder", function() {
    afterEach(cleanup);

    it("works with no count provided", function() {
        const { container } = render(
            <Placeholder />
        );

        expect(container).toMatchSnapshot();
    });

    it("works with a count provided", function() {
        const { container } = render(
            <Placeholder count={3} />
        );

        expect(container).toMatchSnapshot();
    });
});
