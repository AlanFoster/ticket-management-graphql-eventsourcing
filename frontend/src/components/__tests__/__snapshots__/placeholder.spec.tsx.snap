// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder works with a count provided 1`] = `
<div>
  <div
    class="makeStyles-root-4"
  >
    <span
      class="makeStyles-element-6"
    />
    <span
      class="makeStyles-element-6"
    />
    <span
      class="makeStyles-element-6"
    />
  </div>
</div>
`;

exports[`Placeholder works with no count provided 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <span
      class="makeStyles-element-3"
    />
  </div>
</div>
`;
